components/header-bar/src/online-status.styles.js

```diff
@@ -0,0 +1,85 @@
+import { colors, spacers } from '@dhis2/ui-constants'
+import css from 'styled-jsx/css'
+
+export default css`
+    .container {
+        display: flex;
+        align-items: center;
+        justify-content: center; // new
+        background-color: #104167;
+        flex-shrink: 0; // ?
+        color: ${colors.grey100};
+    }
+
+    .container.desktop {
+        margin-right: ${spacers.dp24};
+        padding: ${spacers.dp8};
+        border-radius: 5px;
+        font-size: 14px;
+    }
+
+    .container.mobile {
+        display: none;
+        padding: 0px ${spacers.dp4};
+        min-height: 24px;
+        font-size: 13px;
+    }
+
+    @media (max-width: 480px) {
```

Comment:
For now, let's stick with 480px as our "mobile viewport" limit. This may change in the future, but keeping it as simple as possible first.

Reply:
> it can also get complicated if there's additional text in the badge like 'Last online 10 minutes ago' which takes up a lot of space

Indeed, especially when localizing this string. Without getting into the complicated width-detection, I think the best approach might be to use `ellipsis` for the overflow on the additional text, at least making sure there's no double line-heights from line breaks or anything. This won't solve the problem of too much content, but it makes sure the layout doesn't break.

Reply:
I tried `ellipsis` text overflow and a few other style changes, but in my brief attempt I couldn't get the additional info text to shrink/truncate before the App title text in the header bar shrinks -- I think I'll come back to that goal when additional info gets implemented fully. In the mean time, the info text won't split into multiple lines because of `flex-shrink: 0` on the badge 👍

Reply:
- Added a note to the todos above